File: src/main/java/org/goafabric/invoice/process/adapter/patient/dto/MedicalRecord.java
package org.goafabric.invoice.process.adapter.patient.dto;

import org.goafabric.invoice.process.adapter.patient.type.MedicalRecordType;

public record MedicalRecord(
        String id,
        String encounterId,
        Long version,
        MedicalRecordType type,
        String display,
        String code,
        String specialization) {
    public MedicalRecord(MedicalRecordType type, String display, String code) {
        this(null, null, null, type, display, code, null);
    }

    public MedicalRecord(MedicalRecordType type, String display, String code, String specialization) {
        this(null, null, null, type, display, code, specialization);
    }

}

File: src/main/kotlin/org/goafabric/invoice/process/adapter/patient/dto/MedicalRecord.kt
package org.goafabric.invoice.process.adapter.patient.dto

data class MedicalRecord(
    val id: String?,
    val encounterId: String?,
    val version: Long?,
    val type: MedicalRecordType?,
    val display: String?,
    val code: String?,
    val specialization: String?
)